File: Frontend/library-app/src/models/AddBookRequest.ts
class AddBookRequest {
  title: string;
  author: string;
  description: string;
  copies: number;
  category: string;
  img?: string;
  publisher: string;
  price: number;
  coin: number;
  publicationDate: string;

  constructor(
    title: string,
    author: string,
    description: string,
    copies: number,
    category: string,
    publisher: string,
    price: number,
    coin: number,
    publicationDate: string
  ) {
    this.title = title;
    this.author = author;
    this.description = description;
    this.copies = copies;
    this.category = category;
    this.publisher = publisher;
    this.price = price;
    this.coin = coin;
    this.publicationDate = publicationDate;
  }
}
export default AddBookRequest;

File: src/main/java/com/reactlibraryproject/springbootlibrary/RequestModels/AddBookRequest.java
package com.reactlibraryproject.springbootlibrary.RequestModels;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AddBookRequest {

    private String title;

    private String author;

    private String description;

    private int copies;

    private String category;

    private String img;

    private String publisher;

    private int price;

    private int coin;

    private String publicationDate;

}
